src/System Application/Test/Feature Key/src/FeatureKeyTest.Codeunit.al

```diff
@@ -503,6 +503,63 @@ codeunit 135003 "Feature Key Test"
         asserterror error('') // roll back
     end;
 
+    [Test]
+    procedure T108_IsEnabledAllowInsertTrue()
+    var
+        FeatureDataUpdateStatus: Record "Feature Data Update Status";
+        Enabled: Option "None","All Users";
+        ID: Text[50];
+    begin
+        PermissionsMock.Set('Feature Key Admin');
+        Initialize();
+        ID := 'SalesPrices';
+        // [GIVEN] Feature Key 'X', where "Enabled" is 'All Users' with data update required
+        SetFeatureEnabled(ID, Enabled::"All Users");
+
+        // [GIVEN] FeatureDataUpdateStatus for Company 'A' does not exist
+        FeatureDataUpdateStatus.DeleteAll();
+
+        // [WHEN] run IsEnabled() with AllowInsert = true
+        // [THEN] result is False, because it requires data update
+        Assert.IsFalse(FeatureManagementFacade.IsEnabled(ID, true), 'should be disabled');
+
+        // [THEN] FeatureDataUpdateStatus for Company 'A' is created
+        FeatureDataUpdateStatus.SetRange("Feature Key", ID);
+        FeatureDataUpdateStatus.SetRange("Company Name", CopyStr(CompanyName(), 1, MaxStrLen(FeatureDataUpdateStatus."Company Name")));
+        Assert.RecordIsNotEmpty(FeatureDataUpdateStatus);
+
+        asserterror error('') // roll back
+    end;
+
+    [Test]
+    procedure T109_IsEnabledAllowInsertFalse()
+    var
+        FeatureDataUpdateStatus: Record "Feature Data Update Status";
+        Enabled: Option "None","All Users";
+        ID: Text[50];
+    begin
+        PermissionsMock.Set('Feature Key Admin');
+        Initialize();
+        ID := 'SalesPrices';
+        // [GIVEN] Feature Key 'X', where "Enabled" is 'All Users' with data update required
+        SetFeatureEnabled(ID, Enabled::"All Users");
+
+        // [GIVEN] FeatureDataUpdateStatus for Company 'A' does not exist
+        FeatureDataUpdateStatus.DeleteAll();
+
+        // [WHEN] run IsEnabled() with AllowInsert = false
```

Comment:
This WHEN here doesn't make much sense. 
Makes more sense to me if this is converted to WHEN:
// [GIVEN] FeatureDataUpdateStatus for Company 'A' does not exist

When All Feature Data Update Status is deleted

and then 
// [THEN] result is False, because it requires data update

Reply:
I don't agree, I think current is better.